api/provider/aws/load_balancer_delete.go

```diff
@@ -44,6 +76,15 @@ func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
 		}
 	}
 
+	// Check for eventually consistency
+	fnWaitTillSGDeleted := func() (shouldRetry bool, err error) {
```

Comment:
I think it would be better if `waitTillSGDeleted` actually returned a `func() (shouldRetry bool, err error)`; that would make the callers' lives a little easier, e.g.
```
// in common.go
func waitTillSGDeletedFN(ec2api ec2iface.EC2API, securityGroupName string) func() (shouldRetry bool, err error) {
    return func() (shouldRetry bool, err error) {
        filter := &ec2.Filter{}
        ...
    }
}

// and in this file:
fn := waitTillSGDeletedFN(l.AWS.EC2, securityGroupName)
if err := retry.Retry(fn, retry.WithTimeout(time.Second*30), retry.WithDelay(time.Second)); err != nil {
...
```